### PR TITLE
3582 chartmogul

### PR DESCRIPTION
**Pull request by luancazarine**
ChartMogul

Actions
  - Add a Custom Attribute to a Customer via Email
  - Add a Custom Attribute to a Customer via UUID
  - Add Tags to a Customer via UUID
  - Add Tags to a Customer via Email
  - Create a Customer
  - Merge Customers
  - Update a Customer
  - Search for Customer via Email
  - Search for Customer via External ID
  - Retrieve a Customer Profile
  - Retrieve All Key Metrics
  - Retrieve Average Revenue Per Account (ARPA)
  - Retrieve ARR
  - Retrieve Average Sale Price (ASP)
  - Retrieve Customer Churn Rate
  - Retrieve Customer Count
  - Retrieve Customer Lifetime Value (LTV)
  - Retrieve MRR
  - Retrieve MRR Churn Rate

### CODE REVIEW

File: components/chartmogul/actions/add-custom-attributes-via-email/add-custom-attributes-via-email.mjs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    } = this;
+
+    const response = await this.chartmogul.addInfoViaEmail({
+      $,
+      info: "custom",
+      email: customerEmail,
+      custom: custom.map((item) => JSON.parse(item)),
+    });
+
+    $.export("$summary", "Custom Attributes Successfully added");
```
Comment (lcaresia) [inline components/chartmogul/actions/add-custom-attributes-via-email/add-custom-attributes-via-email.mjs:38]: You can do this to the others actions too.

Obs: I didn't tested this.

```suggestion
    $.export("$summary", `Custom attributes successfully added with ID ${response.id}`);
```

File: components/chartmogul/actions/add-tags-via-email/add-tags-via-email.mjs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import chartmogul from "../../chartmogul.app.mjs";
+
+export default {
+  key: "chartmogul-add-tags-via-email",
+  name: "Add Tags To Customer Via Email",
+  version: "0.0.3",
```
Comment (lcaresia) [inline components/chartmogul/actions/add-tags-via-email/add-tags-via-email.mjs:6]:
```suggestion
  version: "0.0.1",
```

File: components/chartmogul/package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "@pipedream/chartmogul",
+  "version": "0.0.2",
```
Comment (lcaresia) [inline components/chartmogul/package.json:3]:
```suggestion
  "version": "0.0.1",
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Need just some fixes!

**Review by lcaresia — CHANGES_REQUESTED** [review]
Fix the `export` summary to all actions/sources too.

Like this https://github.com/PipedreamHQ/pipedream/pull/5014#discussion_r1055476576

**Review by lcaresia — DISMISSED** [review]
LGTM!

File: components/chartmogul/actions/search-customers-via-email/search-customers-via-email.mjs [hunk 4]
```diff
@@ -19,19 +19,23 @@ export default {
...
-    for await (const item of items) {
-      response.push(item);
+      const response = [];
+      for await (const item of items) {
+        response.push(item);
+      }
+      $.export("$summary", "Customers Successfully fetched");
+      return response.reverse();
+    } catch (e) {
+      return;
```
Comment (alysonturing) [inline components/chartmogul/actions/search-customers-via-email/search-customers-via-email.mjs:38]: you should expose the error for better use experience, in this case, you could log the original error `log(e);` and throw a ConfigurationError with a custom message.

File: components/chartmogul/actions/search-customers-via-external-id/search-customers-via-external-id.mjs [hunk 5]
```diff
@@ -19,20 +19,24 @@ export default {
...
+      for await (const item of items) {
+        response.push(item);
+      }
 
-    $.export("$summary", "Customers Successfully fetched");
-    return response.reverse();
+      $.export("$summary", "Customers Successfully fetched");
+      return response.reverse();
+    } catch (e) {
+      return;
```
Comment (alysonturing) [inline components/chartmogul/actions/search-customers-via-external-id/search-customers-via-external-id.mjs:39]: you should expose the error for better use experience, in this case, you could log the original error log(e); and throw a ConfigurationError with a custom message.

**Review by alysonturing — CHANGES_REQUESTED** [review]
Thanks for your contribution, please check the following warnings and then you're good to go.

**Review by alysonturing — APPROVED** [review]
LGTM!

**Pull request merged**